### DISCUSSION TITLE
Idea: what about typeprovider like f# does

### DISCUSSION THREAD

**Discussion by furesoft** [qwertie/ecsharp#73] [Q&A]
f#'s typeproviders are really cool stuff but i miss it in c# so why not implement it in ec#?

**Comment by qwertie** [comment]
Hi Chris. F# Type Providers are a cool feature and someday I'd like to do something like that. Unfortunately it requires a full [EC# compiler](https://github.com/jonathanvdc/ecsc), which is not nearly ready at this time.

**Comment by qwertie** [comment]
I should add that you can, of course, use EC# as-is (LeMP) to generate classes/structs in various ways. F# type providers are more sophisticated, particularly as they can generate "virtual" types that don't exist at runtime. EC# limits you to creating "real" types and you may have to indicate explicitly which types you want to create (instead of them being created magically when you use them) - even so it's pretty useful.
